Runtime/Triangulator.cs

```diff
@@ -283,49 +288,485 @@ public void Dispose()
-                restoreBoundary = Settings.RestoreBoundary,
-                refineMesh = Settings.RefineMesh,
-                sloanMaxIters = Settings.SloanMaxIters,
-                concentricShellsParameter = Settings.ConcentricShellsParameter,
-                refinementThresholdArea = Settings.RefinementThresholds.Area,
-                refinementThresholdAngle = Settings.RefinementThresholds.Angle,
-                input = new()
-                {
-                    Positions = Input.Positions,
-                    ConstraintEdges = Input.ConstraintEdges,
-                    HoleSeeds = Input.HoleSeeds,
-                },
-                output = Output,
+            return TriangulateAsync(Settings, new() {
+                Positions = Input.Positions,
+                ConstraintEdges = Input.ConstraintEdges,
+                HoleSeeds = Input.HoleSeeds,
+            }, new OutputData<float2> {
+                Positions = outputPositions,
+                Triangles = triangles,
+                Status = status,
+            }, dependencies);
+        }
+
+        public static JobHandle TriangulateAsync(TriangulationSettings settings, InputData<float2> input, OutputData<float2> output, JobHandle dependencies = default)
+        {
+            return new TriangulationJob<float2, float, FloatCoordinateUtils>
+            {
+                preprocessor = settings.Preprocessor,
+                validateInput = settings.ValidateInput,
+                restoreBoundary = settings.RestoreBoundary,
+                refineMesh = settings.RefineMesh,
+                sloanMaxIters = settings.SloanMaxIters,
+                concentricShellsParameter = settings.ConcentricShellsParameter,
+                refinementThresholdArea = (ulong)settings.RefinementThresholds.Area,
+                refinementThresholdAngle = settings.RefinementThresholds.Angle,
+                input = input,
+                output = output
+            }.Schedule(dependencies);
+        }
+
+        public static JobHandle TriangulateAsync(TriangulationSettings settings, InputData<int2> input, OutputData<int2> output, JobHandle dependencies = default)
+        {
+            return new TriangulationJob<int2, ulong, IntCoordinateUtils>
+            {
+                preprocessor = settings.Preprocessor,
+                validateInput = settings.ValidateInput,
+                restoreBoundary = settings.RestoreBoundary,
+                refineMesh = settings.RefineMesh,
+                sloanMaxIters = settings.SloanMaxIters,
+                concentricShellsParameter = settings.ConcentricShellsParameter,
+                refinementThresholdArea = (ulong)settings.RefinementThresholds.Area,
+                refinementThresholdAngle = settings.RefinementThresholds.Angle,
+                input = input,
+                output = output
             }.Schedule(dependencies);
         }
 
+        struct FloatCoordinateUtils : ICoordinateUtils<float2, float> {
```

Comment:
I would move all `ICoordinateUtils<T1, T2>` implementation and the interface itself into `Utils` region.